lib/iex/lib/iex/autocomplete.ex

```diff
@@ -125,6 +128,46 @@ defmodule IEx.Autocomplete do
     end
   end
 
+  defp get_signatures!(name, aliases) when is_list(aliases) do
```

Comment:
```suggestion
  defp get_signatures!(name, module) when is_atom(module) do
```

So this function receives a module and you don't need to call `get_signatures!(name, [module])` below. Also note we already have some functionality for expanding aliases which you should use. See `expand_alias/2`.

Reply:
Sorry I'm not sure I understand: the first `get_signatures!` is also used in `expand_help` when you already have the module after matching `{:__aliases__, _, aliases}`. It's for cases like `IEx.Introspection.open` so you get `[:IEx, :Introspection]` directly into the function call (so you don't need to search for it in the evaluator environment).

Reply:
I mean that you should expand the aliases before you call `get_signature!`. :)

Reply:
ah! sorry now I saw `expand_alias`